arch/Kconfig

```diff
@@ -579,11 +575,12 @@ config IRQ_OFFLOAD
 config IRQ_OFFLOAD_NESTED
 	bool "irq_offload() supports nested IRQs"
 	depends on IRQ_OFFLOAD
+	default y if ARM64 || X86 || RISCV || XTENSA
```

Comment:
The side effect of this is that you can turn it off at build time, e.g. `west build`, which I think is not quite the correct behavior. Maybe we can leave the other arch to select this kconfig. But for Xtensa, this is done in the SoC layer?

Reply:
I think that's sort of the point.  Xtensa is the first arch to hit this edge case.  What's happening here is that we have an architecture which can support nested interrupts, but an instantiation that doesn't actually have a high priority software interrupt set up that irq_offload() can use, so in practice we can't do it.  That's likely to be a situation lots of devices could hit in practice.

Basically it was too hard-coded, and needs to be a little softer IMHO.  And in any case irq_offload() really only exists for test purposes anyway.